server/src/main/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -1183,50 +1221,43 @@ public List<Diagnosis> getDiagnosis(boolean verbose, int maxAffectedResourcesCou
         static List<Diagnosis.Resource> getRestoreFromSnapshotAffectedResources(
             Metadata metadata,
             SystemIndices systemIndices,
-            Set<String> restoreFromSnapshotIndices,
-            int maxAffectedResourcesCount
+            Set<ProjectIndexName> restoreFromSnapshotIndices,
+            int maxAffectedResourcesCount,
+            ProjectResolver projectResolver
```

Comment:
I think I would be leaning towards passing a `boolean supportsMultipleProjects` here instead of a `ProjectResolver`. We only use the resolver for that field in this method. That would also avoid us having to pass project resolvers (and pass booleans instead) in some tests. What do you think?